plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlScannerTest.java

```diff
@@ -17,19 +17,64 @@
 package com.google.cloud.tools.eclipse.appengine.validation;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
 
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+
+import org.eclipse.core.resources.IContainer;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IPackageFragmentRoot;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.core.JavaModelException;
 import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
 import org.junit.Test;
 import org.xml.sax.SAXException;
 import org.xml.sax.ext.Locator2Impl;
 import org.xml.sax.helpers.AttributesImpl;
 
+import com.google.cloud.tools.eclipse.test.util.project.TestProjectCreator;
+
 public class WebXmlScannerTest {
 
-  private WebXmlScanner scanner = new WebXmlScanner();
+  private static IFile resource;
+  private static IFile servletClass;
+  private static IJavaProject javaProject;
+  private WebXmlScanner scanner;
+  
+  @ClassRule public static TestProjectCreator projectCreator = new TestProjectCreator().withClasspathContainerPath("src/main/java");
+  
+  @BeforeClass
+  public static void setUpBeforeClass() throws CoreException {
+    IProject project = projectCreator.getProject();
+    javaProject = JavaCore.create(project);
+    
+    IPackageFragmentRoot root = javaProject.getPackageFragmentRoot("src/main/java");
+    root.createPackageFragment("com.test", false, null);
+    createFolders(project, new Path("src/main/webapp/WEB-INF"));
+    resource = project.getFile("src/main/webapp/WEB-INF/web.xml");
+    resource.create(new ByteArrayInputStream(new byte[0]), true, null);
+    createFolders(project, new Path("src/main/java"));
+    
+    servletClass = project.getFile("src/main/java/ServletClass.java");
+    servletClass.create(new ByteArrayInputStream(
+        "public class ServletClass {}".getBytes(StandardCharsets.UTF_8)), true, null);
```

Comment:
I'd recommend using a different package name. Using `src/main/java` as a source root folder is a Maven convention, which is fine (though as @chanseokoh mentioned, the default in an Eclipse project is `src`. To add the class to a package, I suggest to use a package name like `com.example` or `com.google.cloud.tools.eclipse.testing`.